### PR TITLE
Support to select folder hosts for import Github/Gitlab repo

### PR DESCRIPTION
**Pull request by studioego**
Support to select folder hosts for import Github/Gitlab repository.

If you import Github/Gitlab repository, It only saves default host storage.
However, one of Some customers needs to select some storage
not only default host storage but one of custom host storage.

As a result, We support to select folder hosts for import Github/Gitlab repositories.

related #1278

### CODE REVIEW

**Review by adrysn — CHANGES_REQUESTED** [review]
I think the layout should be more aligned... 

image

Alternatively, I suggest the following:
* Hide the Host from the main page.
* When a user clicks the "GET TO FOLDER" button, display the Host panel as a dialog.
  * For convenience, just use the host without opening the Host dialog when there is only one host.

In the meantime, what is the purpose of the information button? It just displays "No description" message.
image

**Comment by studioego** [comment]
> I think the layout should be more aligned...

I think your alternative option, I decide the first more aligned layout option.

> In the meantime, what is the purpose of the information button? It just displays "No description" message.
That feature copy from 'Data & Storage new storage popup.
image
If you don't want to shows information button, I'll remove information button.

**Comment by studioego** [comment]
Now, I changed the import screen layout. 
image
However, Unlike Import Gitlab repository layout, To set Github repository layout 
is difficult for me.
I will think about a better layout.

**Comment by studioego** [comment]
Fix layout!
Below is fixed import layout. 
image

**Review by studioego — COMMENTED** [review]
Following your opinion, I changed both Github and Gitlab layouts.
However, Your alternative option should be considered. 
(I have to learn more typescript and modern html, etc)

**Review by adrysn — CHANGES_REQUESTED** [review]
The updated layout looks as you intended in the Korean page but not in the English page. We need to think about a way to align those elements in any languages.

image

**Comment by studioego** [comment]
> The updated layout looks as you intended in the Korean page but not in the English page. We need to think about a way to align those elements in any languages.
> 
> image

I got it. I'll try to resolve the layout issue.

**Comment by studioego** [comment]
I checked the import layout on i18n environment ( sucn as English, Bahasa Indonesia, etc)

* English 

image

* Korean
image


* Bahasa Indonesia
image

* Mongolian
image


* Russian

image

* French
image

File: src/components/backend-ai-import-view.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@
 import {get as _text, translate as _t} from 'lit-translate';
 import {css, CSSResultGroup, html} from 'lit';
 import {customElement, property} from 'lit/decorators.js';
-
+import {unsafeHTML} from 'lit/directives/unsafe-html.js';
```
Comment (adrysn) [inline src/components/backend-ai-import-view.ts]: I don't see any usage of the `unsafeHTML` in this PR. If it is not used,please remove it.

Comment (adrysn) [inline src/components/backend-ai-import-view.ts]: I don't see any usage of the `unsafeHTML` in this PR. If it is not used,please remove it.

File: src/components/backend-ai-import-view.ts [hunk 2]
```diff
@@ -80,22 +89,73 @@ export default class BackendAIImport extends BackendAIPage {
...
-        mwc-textfield#github-repo-url {
+        mwc-textfield#notebook-url {
           width: 75%;
         }
+        mwc-textfield#github-repo-url {
+          width: 100%;
+        }
+        mwc-textfield#gitlab-repo-url {
+          width: 100%;
+        }
```
Comment (adrysn) [inline src/components/backend-ai-import-view.ts]: Let's merge these blocks into:
```javascript
mwc-textfield#github-repo-url,
mwc-textfield#gitlab-repo-url {
  width: 100%;
}
```

Or, just tag the same class, such as `repo-url`, to both of the elements and:
```javascript
mwc-textfield.repo-url {
  ...
} 
```

Comment (adrysn) [inline src/components/backend-ai-import-view.ts]: Let's merge these blocks into:
```javascript
mwc-textfield#github-repo-url,
mwc-textfield#gitlab-repo-url {
  width: 100%;
}
```

Or, just tag the same class, such as `repo-url`, to both of the elements and:
```javascript
mwc-textfield.repo-url {
  ...
} 
```

File: src/components/backend-ai-import-view.ts [hunk 3]
```diff
@@ -481,6 +581,37 @@ export default class BackendAIImport extends BackendAIPage {
...
+      this._helpDescription = this.storageInfo[item].description;
+      this._helpDescriptionIcon = this.storageInfo[item].icon;
+    } else {
+      this._helpDescriptionTitle = item;
+      this._helpDescriptionIcon = 'local.png';
+      this._helpDescription = _text('data.NoStorageDescriptionFound');
+    }
+    const desc = this.shadowRoot.querySelector('#help-description');
+    desc.show();
+  }
```
Comment (adrysn) [inline src/components/backend-ai-import-view.ts]: If not use, please remove it.

Comment (adrysn) [inline src/components/backend-ai-import-view.ts]: If not use, please remove it.

**Review by adrysn — CHANGES_REQUESTED** [review]
The features are working fine for me. I found minor issues in styles and not-used code blocks, and if they are fixed, I think I can merge this PR.

File: src/components/backend-ai-import-view.ts (hunk 3 above)
Comment (studioego) [reply src/components/backend-ai-import-view.ts]: delete it

File: src/components/backend-ai-import-view.ts (hunk 1 above)
Comment (studioego) [reply src/components/backend-ai-import-view.ts]: delete it

File: src/components/backend-ai-import-view.ts (hunk 2 above)
Comment (studioego) [reply src/components/backend-ai-import-view.ts]: change the `repo-url` tags

**Review by studioego — COMMENTED** [review]
Applyed

**Review by adrysn — APPROVED** [review]
LGTM.

**Pull request merged**